projects/plugins/crm/src/automation/commons/conditions/contacts/class-contact-field-changed.php

```diff
@@ -18,26 +19,37 @@
 class Contact_Field_Changed extends Base_Condition {
 
 	/**
-	 * All valid operators for this condition.
+	 * Contact_Field_Changed constructor.
 	 *
 	 * @since $$next-version$$
-	 * @var string[] $valid_operators Valid operators.
-	 */
-	protected $valid_operators = array(
-		'is',
-		'is_not',
-	);
-
-	/**
-	 * All valid attributes for this condition.
 	 *
-	 * @since $$next-version$$
-	 * @var string[] $valid_operators Valid attributes.
+	 * @param array $step_data The step data.
 	 */
-	private $valid_attributes = array(
-		'operator',
-		'value',
-	);
+	public function __construct( array $step_data ) {
+		parent::__construct( $step_data );
+
+		// TODO: Fetch automation fields from our DAL.
+		$contact_fields = array(
+			'id'       => __( 'ID', 'zero-bs-crm' ),
+			'fname'    => __( 'First Name', 'zero-bs-crm' ),
+			'lname'    => __( 'Last Name', 'zero-bs-crm' ),
+			'fullname' => __( 'Full Name', 'zero-bs-crm' ),
+		);
+
+		$this->valid_operators = array(
```

Comment:
I wonder if operators such as these could be declared and reused from somewhere so that the copy for them doesn't have to be repeated in each condition.

Reply:
I had the same feeling. I will try to improve this!